### PR TITLE
feat: add "headless" json-rpc based REPL

### PR DESCRIPTION
**Pull request by Westonside**
## Description
Added in the existing code for the repl and modified existing repl methods so that the communication was done via the stdin and out. Related to : #4793 .

## Input Example

``` {"jsonrpc":"2.0","method":"Service.DidOutput","params":[{"input":"x=1"}]}```

### Checklist

Dear Author :wave:, the following checks should be completed (or explicitly dismissed) before merging.

- [x] ✏️ Write a PR description, regardless of triviality, to include the _value_ of this PR
- [x] 🔗 Reference related issues
- [x] 🏃 Test cases are included to exercise the new code
- [x] 🧪 If **new packages** are being introduced to stdlib, link to Working Group discussion notes and ensure it lands under `experimental/`
- [x] 📖 If **language features** are changing, ensure `docs/Spec.md` has been updated

Dear Reviewer(s) :wave:, you are responsible (among others) for ensuring the completeness and quality of the above before approval.

### CODE REVIEW

File: internal/cmd/headless_repl/main.go [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+	return err
+}
+
+type Response struct {
+	Result string
+}
+
+type Testing struct {
+	A string `json:"input"`
+}
```
[the remark this answers is not in the document]
Comment (Westonside) [reply internal/cmd/headless_repl/main.go:123]: Sorry, I forgot to commit the changes for these deletions so things were a little behind.

File: internal/cmd/headless_repl/main.go [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+	r.setCancel(nil)
+}
+
+func (r *ScopeHolder) Input(t string) (*libflux.FluxError, error) {
+	a, err := r.executeLine(t)
+	return a, err
+}
+
+// input processes a line of input and prints the result.
+func (r *ScopeHolder) input(t string) {
```
[the remark this answers is not in the document]
Comment (Westonside) [reply internal/cmd/headless_repl/main.go:210]: I am adding in that name  change, but the naming is the same in the current repl as well.

File: internal/cmd/headless_repl/main.go [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+		}
+	}
+	return dirs, nil
+}
+
+// LoadQuery returns the Flux query q, except for two special cases:
+// if q is exactly "-", the query will be read from stdin;
+// and if the first character of q is "@",
+// the @ prefix is removed and the contents of the file specified by the rest of q are returned.
+func LoadQuery(q string) (string, error) {
```
[the remark this answers is not in the document]
Comment (Westonside) [reply internal/cmd/headless_repl/main.go:368]: I kept this implementation from the original repl, so I am assuming this was intended.

**Review by onelson — CHANGES_REQUESTED** [review]
Looks like CI is failing for `make staticcheck` which reports some unused code in here.

Once CI is passing, I think this is good to merge.

**Review by onelson — APPROVED** [review]
Looking good! Time to merge.

**Pull request merged**